bundles/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/coe/TACmiCoEBridgeHandler.java

```diff
@@ -0,0 +1,250 @@
+                    } else if (data[1] == 0 || data[1] == 9) {
+                        message = new DigitalMessage(data);
+                    } else {
+                        logger.debug("Invalid message received");
+                        continue;
+                    }
+                    logger.debug("{}", message.toString());
+
+                    final InetAddress remoteAddress = receivePacket.getAddress();
+                    final int node = message.canNode;
+                    boolean found = false;
+                    for (final TACmiHandler cmi : registeredCMIs) {
+                        if (cmi.isFor(remoteAddress, node)) {
+                            cmi.handleCoE(message);
+                            found = true;
+                        }
+                    }
+                    if (!found) {
+                        logger.debug("Received CoE-Packet from {} Node {} and we don't have a Thing for!",
+                                remoteAddress, node);
+                    }
+                } catch (final IOException e) {
+                    if (isInterrupted()) {
+                        return;
+                    }
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Error processing data: " + e.getMessage());
+
+                } catch (RuntimeException e) {
+                    // we catch runtime exceptions here to prevent the receiving thread to stop accidentally if
+                    // something like a IllegalStateException or NumberFormatExceptions are thrown. This indicates a bug
+                    // or a situation / setup I'm not thinking of ;)
+                    if (isInterrupted()) {
+                        return;
+                    }
+                    logger.error("Error processing data: {}", e.getMessage(), e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Periodically check for timeouts on the registered / active CoE channels
+     */
+    private void checkForTimeouts() {
+        for (final TACmiHandler cmi : registeredCMIs) {
+            cmi.checkForTimeout();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            final DatagramSocket coeSocket = new DatagramSocket(coePort);
+            coeSocket.setBroadcast(true);
+            coeSocket.setSoTimeout(330000); // 300 sec is default resent-time; so we wait 330 secs
+            this.coeSocket = coeSocket;
+        } catch (final SocketException e) {
+            // logged by framework via updateStatus
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Failed to create UDP-Socket for C.M.I. CoE bridge. Reason: " + e.getMessage());
+            return;
+        }
+
+        ReceiveThread reciveThreadNN = new ReceiveThread(
+                "OH-" + TACmiBindingConstants.BINDING_ID + "-" + getThing().getUID().getAsString());
```

Comment:
The name should not contain your binding name, but the string "binding".

Reply:
fixed...